runtime/src/accounts_db.rs

```diff
@@ -1145,13 +1222,26 @@ impl AccountsDB {
         slot: Slot,
         accounts: &[(&Pubkey, &Account)],
         hashes: &[Hash],
+    ) -> Vec<AccountInfo> {
+        let write_version = self.write_version.fetch_add(accounts.len(), Ordering::Relaxed) as u64;
```

Comment:
The write_version was unique before for all accounts, now you can have overlapping ones if multiple threads are calling the store. I'm not sure why this change is necessary.

Reply:
Well, I thought these could be valid optimization. Because these are atomic, I think there is no possible case multiple threads could be given overlapping ranges.

Say, Thread A has 100 updates and write_version is 3, then it set 103 and uses [3..103) and write_version atomically updated to 103. Then thread b has 200 updates and write_version is 103. Then, it sets 303 and uses [103..303)....

Ouch, impl is wrong.. I forgot to substract...

Anyway, this is semantically just  batching of fetch_add(+1). So, this should be problem and can avoid too much memory contention? Am I missing something? Maybe? I just noticed and jot down not to forget. :)

Reply:
Ah, I didn't see that it does `fetch_add(accounts.len())`. Ok, yea this should be fine, although this might be 0.1% of the time of the work in this function.

Reply:
Wow, `0.1%`... sad. Didn't measured. ;)

Reply:
Just a guess :)